### PR TITLE
DateTimeAdapter tests

### PR DESCRIPTION
**Pull request by DeanWay**

### CODE REVIEW

**Comment by DeanWay** [comment]
@JKereliuk @tmarathe take a look. Any suggestions?

**Comment by DeanWay** [comment]
Forgot licence header. Will fix that

**Comment by tmarathe** [comment]
:+1:

File: components/patient-data/rest/src/test/java/org/phenotips/data/rest/DateTimeAdapterTest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+/**
+ * Test for the {@link DateTimeAdapter} class
```
Comment (sdumitriu) [inline components/patient-data/rest/src/test/java/org/phenotips/data/rest/DateTimeAdapterTest.java]: End sentences with a dot.

File: components/patient-data/rest/src/test/java/org/phenotips/data/rest/DateTimeAdapterTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ */
+public class DateTimeAdapterTest {
```
Comment (sdumitriu) [inline components/patient-data/rest/src/test/java/org/phenotips/data/rest/DateTimeAdapterTest.java]: Doesn't follow the [codestyle](https://github.com/phenotips/build-tools/tree/master/codestyle-resources/src/main/resources).

File: components/patient-data/rest/src/test/java/org/phenotips/data/rest/DateTimeAdapterTest.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    private DateTimeAdapter adapter;
+
+    private DateTime dateTime;
+
+    private String ValidDateString;
```
Comment (sdumitriu) [inline components/patient-data/rest/src/test/java/org/phenotips/data/rest/DateTimeAdapterTest.java]: Instance field must start with lowercase letter.

File: components/patient-data/rest/src/test/java/org/phenotips/data/rest/DateTimeAdapterTest.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    public void marshalReturnsCorrectString() throws Exception
+    {
+        Assert.assertEquals(this.ValidDateString, this.adapter.marshal(this.dateTime));
+    }
+
+    @Ignore
+    @Test(expected = Exception.class)
+    public void marshalThrowsExceptionWhenPassedNull() throws Exception
+    {
+        this.adapter.marshal(null);
```
Comment (sdumitriu) [inline components/patient-data/rest/src/test/java/org/phenotips/data/rest/DateTimeAdapterTest.java]: Code has been updated, you must update the test as well. Now `null` is accepted.

File: components/patient-data/rest/src/test/java/org/phenotips/data/rest/DateTimeAdapterTest.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    @Test(expected = Exception.class)
+    public void unMarshalThrowsExceptionWhenPassedInvalidString() throws Exception
+    {
+        this.adapter.unmarshal("2015-02-29T12:00:00.000Z");
+    }
+
+    @Test(expected = Exception.class)
+    public void unMarshalThrowsExceptionWhenPassedEmptyString() throws Exception
+    {
+        this.adapter.unmarshal("");
```
Comment (sdumitriu) [inline components/patient-data/rest/src/test/java/org/phenotips/data/rest/DateTimeAdapterTest.java]: Code has been updated, you must update the test as well. Now `null` and empty strings are accepted.

File: components/patient-data/rest/src/test/java/org/phenotips/data/rest/DateTimeAdapterTest.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+    private String ValidDateString;
+
+
+    @Before
+    public void setUp() throws Exception
+    {
+        this.adapter = new DateTimeAdapter();
+        this.dateTime = new DateTime(2015, 1, 1, 12, 0, DateTimeZone.UTC);
+        this.ValidDateString = "2015-01-01T12:00:00.000Z";
```
Comment (sdumitriu) [inline components/patient-data/rest/src/test/java/org/phenotips/data/rest/DateTimeAdapterTest.java]: I'd like to see some tests using a different timezone.

Comment (DeanWay) [inline components/patient-data/rest/src/test/java/org/phenotips/data/rest/DateTimeAdapterTest.java]: If `marshal()` translates it output base on the UTC timezone, shouldn't `unMarshal()` do the same? Or is that not necessary? 
i.e.:
```java
return DateTime.parse(v, ISO_DATETIME_FORMATTER);
```

Comment (DeanWay) [inline components/patient-data/rest/src/test/java/org/phenotips/data/rest/DateTimeAdapterTest.java]: [It is currently written like this](https://github.com/phenotips/phenotips/blob/master/components/patient-data/rest/src/main/java/org/phenotips/data/rest/DateTimeAdapter.java#L46)

**Pull request merged**